lib/src/algorithms.dart

```diff
@@ -394,8 +394,9 @@ void _merge<E, K>(
   }
   // First list empties first. Reached by break above.
   target[targetOffset++] = secondElement;
-  target.setRange(
-      targetOffset, targetOffset + (secondEnd - cursor2), secondList, cursor2);
+  for (var i = targetOffset; i < targetOffset + (secondEnd - cursor2); i++) {
```

Comment:
Yes, I missed that there are a couple code paths where this could hit a moving insertion sort. I'm not sure how to construct  an input list to exercise that path.

I'll go with the `sublist` approach to fix them reliably.